pkg/agent/run.go

```diff
@@ -257,9 +256,10 @@ func configureNode(ctx context.Context, agentConfig *daemonconfig.Agent, nodes v
 					continue
 				}
 			}
-			logrus.Infof("labels have been set successfully on node: %s", agentConfig.NodeName)
+			logrus.Infof("annotations and labels have been set successfully on node: %s", agentConfig.NodeName)
+			logrus.Infof("new annotations: %v", node.Annotations)
```

Comment:
Can you talk about this change to use annotations?

Reply:
I'm not the author, but it's probably related to the issue described here: https://github.com/k3s-io/k3s/pull/1256#issue-358325912

> This is because IPv6 addresses contain colons (:) and the k3s cloud provider
Implementation stores node's IP addresses as values in the node's
metadata.labels, however Kubernetes doesn't allow colons in label values. In
fact Kubernetes has some strict requirements on label values:
Valid label values must be 63 characters or less and must be empty or begin and end with an alphanumeric character ([a-z0-9A-Z]) with dashes (-), underscores (_), dots (.), and alphanumerics between.

Reply:
That may be a little problematic. The kubelet is allowed to set labels when it registers, and our embedded cloud controller looks at those labels and then sets the node external IP field, as that is a field that the node is not allowed to set for itself. As far as I know the Kubelet is NOT allowed to set its own annotations, so they cannot be used to pass through information in this fashion.

Reply:
I commented this in the PR description, in *Further Comments*

Reply:
@brandond I can confirm the annotations are present in the nodes, is that what you are concerned about?
```
$ sudo kubectl describe node REDACTED_NODENAME
Name:               REDACTED_NODENAME
Roles:              control-plane,master
Labels:             beta.kubernetes.io/arch=amd64
                    beta.kubernetes.io/instance-type=k3s
                    beta.kubernetes.io/os=linux
                    kubernetes.io/arch=amd64
                    kubernetes.io/hostname=REDACTED_NODENAME
                    kubernetes.io/os=linux
                    node-role.kubernetes.io/control-plane=true
                    node-role.kubernetes.io/master=true
                    node.kubernetes.io/instance-type=k3s
Annotations:        io.cilium.network.ipv4-cilium-host: 100.64.0.228
                    io.cilium.network.ipv4-health-ip: 100.64.0.20
                    io.cilium.network.ipv4-pod-cidr: 100.64.0.0/24
                    io.cilium.network.ipv6-cilium-host: fd00:5000::ecd3
                    io.cilium.network.ipv6-health-ip: fd00:5000::35b9
                    io.cilium.network.ipv6-pod-cidr: fd00:5000::/64
                    k3s.io/hostname: REDACTED_NODENAME
                    k3s.io/internal-ip: REDACTED_IPv4,REDACTED_IPv6
                    node.alpha.kubernetes.io/ttl: 0
                    volumes.kubernetes.io/controller-managed-attach-detach: true
[...]
```

Reply:
Interesting, I will admit that I haven't looked at it recently but I thought that nodes didn't have permission to update their own annotations.